### PR TITLE
Workaround a Qt bug on macOS Big Sur

### LINKED ISSUE #30681: MacOS BigSur mantid will not open
**Original reporter:** [Mantid Help Forum](https://forum.mantidproject.org/t/mantid-crash-on-startup-in-big-sur-macos-11-0-1/707/14) and @gvardany :https://github.com/mantidproject/mantid/issues/30642#issuecomment-776520506

After upgrading to MacOS Big Sur workbench packages for 5.1.1 and the release candidate for 6.0 both do not open.

Release 5.0 opens and locally built packages have worked too. This seems to be a Qt version issue: https://bugreports.qt.io/browse/QTBUG-87014

@martyngigg

### PR DESCRIPTION
**Pull request by martyngigg**
**Description of work.**

The fix for Qt has yet to released so we need
to force layer-backed windows on Big Sur
and above

**To test:**

* Build a package on the CI servers
* Test the built package starts on Big Sur

Fixes #30681

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/applications/workbench/workbench/app/start.py [hunk 1]
```diff
@@ -86,6 +86,15 @@ def initialize():
...
         asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())
+    if sys.platform == 'darwin':
+        # Force layer-backing on macOS >= Big Sur (10.16)
+        # or the application hangs
+        # A fix inside Qt is yet to be released:
+        # https://codereview.qt-project.org/gitweb?p=qt/qtbase.git;a=commitdiff;h=c5d904639dbd690a36306e2b455610029704d821
+        from distutils.version import LooseVersion
+        import platform
+        if LooseVersion(platform.mac_ver()[0]) >= '11':
+            os.environ['QT_MAC_WANTS_LAYER'] = '1'
```
Comment (DanielMurphy22) [inline qt/applications/workbench/workbench/app/start.py:97]: Although I'm on version 11, I do not enter this if statement on my computer.

**Review by DanielMurphy22 — CHANGES_REQUESTED** [review]
Built package and it doesn't work for me. Looks like there's another fun bug to "work around": https://stackoverflow.com/questions/65290242/pythons-platform-mac-ver-reports-incorrect-macos-version/65402241

As per the top answer on StackOverflow, I get back the version 10.16 even though I'm on version 11.2

File: qt/applications/workbench/workbench/app/start.py (hunk 1 above)
Comment (martyngigg) [reply qt/applications/workbench/workbench/app/start.py:97]: Thanks for finding this. Time for a workaournd on top of the workaround :(

I think we'll check for both 11 & 10.16 and that should cover it.

**Review by DanielMurphy22 — DISMISSED** [review]
It works now for me! Very good!

**Comment by martyngigg** [comment]
Thanks for testing this. I had to resolve a merge conflict - could you reapprove?

**Review by DanielMurphy22 — APPROVED** [review]
Thanks for fixing the conflict, ready to go!

**Review by DavidFair — APPROVED** [review]

**Pull request merged**